packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -646,6 +649,9 @@ class _PopupMenuRouteLayout extends SingleChildLayoutDelegate {
   // The padding of unsafe area.
   EdgeInsets padding;
 
+  // List of rectangles that we should avoid overlapping. Unusable screen area.
+  final Set<Rect> avoidBounds;
```

Comment:
Does this work in the case of phones that do not have an interrupting hinge in the display? I imagine if someone had a continuous screen across the fold, they would not want to have it show up in the crease.

Reply:
No, and this was intentional. If the screen is continuous, using the crease area as a layout boundary is not required, but rather recommended. 

Example: You could have a `PopupMenuButton` on the crease, and in that case having the popup move to one side might be awkward.

Proposal: It could also make sense to avoid these display features when the posture of the device is half-opened but ignore then when the posture is flat. In those cases, the hinge or crease does separate the screen into sub-screens.

Whatever we decide here might make sense to bring to `DisplayFeatureSubScreen` and the dialogs that use it in `https://github.com/flutter/flutter/pull/92909`. The [material docs](https://m3.material.io/foundations/adaptive-design/foldables/overview) do say to move dialogs to one side, as you suggest, and don't make any special note regarding type of foldable or posture.